pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -0,0 +1,83 @@
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/genevalogging"
+	"github.com/Azure/ARO-RP/pkg/util/dns"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+const (
+	certificateExpirationMetricName = "certificate.expirationdate"
+	secretMissingMetricName         = "certificate.secretnotfound"
+)
+
+func (mon *Monitor) emitCertificateExpirationStatuses(ctx context.Context) error {
+	// report NotAfter dates for Geneva (always), Ingress, and API (on managed domain) certificates
+	var certs []*x509.Certificate
+
+	mdsdCert, err := mon.getCertificate(ctx, operator.SecretName, operator.Namespace, genevalogging.GenevaCertName)
+	if kerrors.IsNotFound(err) {
+		mon.emitGauge(secretMissingMetricName, int64(1), map[string]string{
+			"secretMissing": operator.SecretName,
+		})
+	} else if err != nil {
+		return err
```

Comment:
nit: 
I think in other examples, we nest this.  it allows a go idiomatic approach to return early if there's an error.  

```suggestion
	if err != nil {
            if !kerrors.IsNotFound(err) {
		return err
	    }
	    mon.emitGauge(secretMissingMetricName, int64(1), map[string]string{
			"secretMissing": operator.SecretName,
	    })
	}